File: TD/TD6/Code/P9.cpp
/* Calculer et retourner les N premiers termes de la suite */
#include <iostream>

/* Procédure */

void suite(float tab[], int n)
{
  tab[0] = 1;
  
  for (int i = 1; i < n - 1; ++i) {
    tab[i] = tab[i - 1] / i;
  }
}

/* Utilisation */

int main()
{
  float tab[6] = {};
  
  suite(tab, 6);
  
  for (int i = 0; i < 6; ++i) {
    std::cout << tab[i] << " " << std::flush;
  }
  
  std::cout << std::endl;
  return 0;
}

File: TD/TD6/Code/Python/P9.py
#!/usr/bin/python
# Calculer et retourner les N premiers termes de la suite


def suite(tab):
    tab[0] = 1

    for i in range(1, 5):
        tab[i] = tab[i - 1] / i


tab = [[] for i in range(5)]

suite(tab)

print(tab)
